share_extension/android/selectors/index.js

```diff
@@ -74,14 +78,18 @@ export const getExtensionSortedPrivateChannels = createSelector(
         }
 
         const locale = currentUser.locale || 'en';
-        const publicChannels = teamChannelIds.filter((id) => {
+        return teamChannelIds.reduce((publicChannels, id) => {
             if (!myMembers[id]) {
-                return false;
+                return publicChannels;
             }
+
             const channel = channels[id];
-            return teamChannelIds.includes(id) && channel.type === General.PRIVATE_CHANNEL;
-        }).map((id) => channels[id]).sort(sortChannelsByDisplayName.bind(null, locale));
-        return publicChannels;
+            if (channel.type === General.PRIVATE_CHANNEL) {
+                publicChannels.push(channel);
+            }
+
+            return publicChannels;
+        }, []).sort(sortChannelsByDisplayName.bind(null, locale));
```

Comment:
Non-blocking but just like to mention that `getExtensionSortedPublicChannels` and `getExtensionSortedPrivateChannels` are almost identical.  Maybe pass a second param to that selector as channel type so that we can come up with a generic function?